custom_reg_form/management/commands/email_opt_in_list_to_s3.py

```diff
@@ -0,0 +1,257 @@
+        - you can specify which settings to use for aws access key/secret
+        like so: "--aws-access-key-setting=CVS_EXPORTER_S3_KEY" and using "--aws-secret-key-setting"
+    - other options:
+        - "--skip-when-null", skips the row when the value of one of the fields is null
+"""
+
+import logging
+import csv
+import io
+import tempfile
+import importlib
+import os
+
+import boto3
+
+from django.core.management.base import BaseCommand
+from django.contrib.auth.models import User
+
+from django.db import connections
+from django.conf import settings
+from django.forms import ModelForm
+from botocore.exceptions import NoCredentialsError
+
+from common.djangoapps.student.models import UserProfile
+from common.djangoapps.util.query import read_replica_or_default
+
+
+LOGGER = logging.getLogger(__name__)
+
+def get_registration_ext_model():
+    """Returns the REGISTRATION_EXTENSION_MODEL"""
+    # first attempt to infer from the Meta field in the form
+    if hasattr(settings, "REGISTRATION_EXTENSION_FORM"):
+        try:
+            form_class_path = settings.REGISTRATION_EXTENSION_FORM
+            module_path = ".".join(form_class_path.split(".")[:-1])
+            module = importlib.import_module(module_path)
+            form_class = getattr(module, form_class_path.split(".")[-1])
+            if issubclass(form_class, ModelForm):
+                model_class = form_class.Meta.model
+                return model_class
+        except Exception:   # pylint: disable=broad-except
+            # we ignore execptions so we can try to get the
+            # model class in another way
+            pass
+
+    # check REGISTRATOIN_EXTENSION_MODEL setting
+    if hasattr(settings, "REGISTRATION_EXTENSION_MODEL"):
+        try:
+            model_class_path = settings.REGISTRATION_EXTENSION_MODEL
+            module_path = ".".join(model_class_path.split(".")[:-1])
+            module = importlib.import_module(module_path)
+            model_class = getattr(module, model_class_path.split(".")[-1])
+            return model_class
+        except Exception:    # pylint: disable=broad-except
+            # ignore and try again
+            pass
+
+    # check REGISTRATION_EXTENSION_MODEL environment variable
+    if "REGISTRATION_EXTENSION_MODEL" in os.environ:
+        try:
+            model_class_path = os.environ.get("REGISTRATION_EXTENSION_MODEL")
+            module_path = ".".join(model_class_path.split(".")[:-1])
+            module = importlib.import_module(module_path)
+            model_class = getattr(module, model_class_path.split(".")[-1])
+            return model_class
+        except Exception:  # pylint: disable=broad-except
+            pass
+
+    raise Exception("Couldn't infer the model path to the registration extension model.")
+
+def infer_field_model(field):
+    """Returns the model of which `field` is a member"""
+
+    # first check in the User model
+    if hasattr(User, field):
```

Comment:
Although in 99.9% of the cases it will be the `User` model, this is the place where we should use the swappable user model as per the Django recommendations.

Reply:
Updated!